helix-term/src/commands.rs

```diff
@@ -1668,6 +1668,36 @@ mod cmd {
         Ok(())
     }
 
+    fn buffer_close(
+        cx: &mut compositor::Context,
+        _args: &[&str],
+        _event: PromptEvent,
+    ) -> anyhow::Result<()> {
+        let view = view!(cx.editor);
+        let cur_doc_id = view.doc;
+
+        if let Some(alt) = view.last_accessed_doc {
+            cx.editor.switch(alt, Action::Replace);
+        } else if cx.editor.documents.len() > 1 {
+            let first_other_doc = cx
+                .editor
+                .documents
+                .iter()
+                .find_map(|(&id, _)| if id == cur_doc_id { None } else { Some(id) })
+                .unwrap();
+            cx.editor.switch(first_other_doc, Action::Replace);
+        } else {
+            cx.editor.new_file(Action::Replace);
+        }
+
+        let view_mut = view_mut!(cx.editor);
+        view_mut.jumps.jumps.retain(|(id, _)| id != &cur_doc_id);
+        view_mut.last_accessed_doc = None;
+        cx.editor.documents.remove(&cur_doc_id);
```

Comment:
If there's a language server attached to the doc, you also want to emit a `textDocument/close` event: https://github.com/helix-editor/helix/blob/ded1ac9bfb3d350fa17c5856ae29302a0ea19bbb/helix-view/src/editor.rs#L390-L392

Reply:
Also now that this method exists, let's remove the `close_buffer` parameter from`editor.close` https://github.com/helix-editor/helix/blob/ded1ac9bfb3d350fa17c5856ae29302a0ea19bbb/helix-view/src/editor.rs#L386-L394

It was never used and was implemented incorrectly

Reply:
Funnily enough, my first implementation was attempting to use that... Imagine my surprise when it just exited. I suppose it *did* close the buffer, though... :laughing: